src/data_loader.py

```diff
@@ -113,11 +108,10 @@ def get_ripple_set(args, kg, user_history_dict):
                 tails_of_last_hop = ripple_set[user][-1][2]
 
             for entity in tails_of_last_hop:
-                if entity in kg:
-                    n_neighbor = len(kg[entity])
-                    memories_h.extend(n_neighbor * [entity])
-                    memories_r.extend(kg[entity][i][1] for i in range(n_neighbor))
-                    memories_t.extend(kg[entity][i][0] for i in range(n_neighbor))
+                for tail_and_relation in kg[entity]:
```

Comment:
Entity may not be in kg.  Checks should be made before using kg[entity].

Reply:
This commit makes `kg` a `defaultdict`. Even if `entity` is not inside `kg`, `kg[entity]` will return `[]`.

And `for tail_and_relation in kg[entity]:` will just turn into `for tail_and_relation in []:` - which should be fine.

I will agree that it's not immediately obvious, though. If you prefer it, I can add the check.